tests/test_browser.py

```diff
@@ -2464,12 +2464,12 @@ def test_mmap_file(self):
   def test_emrun_info(self):
     if not has_browser():
       self.skipTest('need a browser')
-    result = run_process([PYTHON, path_from_root('emrun'), '--system_info', '--browser_info'], stdout=PIPE).stdout
+    result = run_process([PYTHON, path_from_root('emrun.py'), '--system_info', '--browser_info'], stdout=PIPE).stdout
```

Comment:
is this change necessary? if it is that seems like a breaking change. if it isn't then why?

Reply:
Yes this change is a slightly breaking change.    You can no longer run `python emcc` (because emcc is now itself a shell script that runs python).   You can still run `./emcc.py` directly or `python emcc.py` if you like.  

I think this is the best approach.     The python_selector method where python re-invokes itself is kind of nasty.   If I run `python2 emcc` I kind of expect it to run under python2 not invoke a python3 subprocess to run itself.

Reply:
sgtm. is there some way we can show a clear error on `python emcc / emranlib / etc.`, like keeping the files around but only containing a print of an error? or if that's not possible, maybe a comment?

Reply:
I'm not deleting those files.  They are now symbolic links to `run_python.sh`.   So they exist by they are shell scripts.

I'm not sure we can have `python <some_shell_script>` do anything useful... but let me see...